apt_info.py

```diff
@@ -60,9 +61,11 @@ def _convert_candidates_to_upgrade_infos(candidates):
     return changes_list
 
 
-def _write_pending_upgrades(registry, cache, logger):
+def _write_pending_upgrades(registry, cache, logger, cli_args):
```

Comment:
again maybe nitpicking, but i would pass the exclude flag directly here, not the whole args namespace:

```python
def _write_pending_upgrades(registry, cache, logger, exclude):
```